File: core/src/main/java/com/github/fburato/highwheelmodules/core/analysis/AnalyserException.java
package com.github.fburato.highwheelmodules.core.analysis;

public class AnalyserException extends RuntimeException {
    private static final long serialVersionUID = -1769077848917606716L;

    public AnalyserException(String msg) {
        super(msg);
    }

    public AnalyserException(Exception cause) {
        super(cause);
    }
}

File: core/src/main/scala/com/github/fburato/highwheelmodules/core/analysis/AnalyserException.scala
package com.github.fburato.highwheelmodules.core.analysis

case class AnalyserException(msg: String, cause: Exception) extends RuntimeException(msg, cause)

object AnalyserException {
  def apply(msg: String): AnalyserException = AnalyserException(msg, null)

  def apply(cause: Exception): AnalyserException = AnalyserException(null, cause)
}
